servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/ReadOnlyHttpServerConfig.java

```diff
@@ -20,50 +20,20 @@
 
 import javax.annotation.Nullable;
 
-final class ReadOnlyHttpServerConfig {
+interface ReadOnlyHttpServerConfig {
```

Comment:
Had to refactor the readonly configs into interfaces so that I could add a delegate in the OptionalSslNegotiator.

NOTE: I know that this is a little incomplete (the client is not converted, and javadocs are missing). I can do some more polish and cleanup in a follow up commit.